Tests/test_inventory.gd

```diff
@@ -0,0 +1,58 @@
+extends GutTest
+
+func before_each() -> void:
+	InventoryManager.reset_inventory()
+
+func test_add_gold() -> void:
+	InventoryManager.gold_component.add_gold(10)
+	assert_eq(InventoryManager.gold_component.get_gold_amount(), 10)
+
+func test_lose_gold() -> void:
+	InventoryManager.gold_component.lose_gold(10)
+	assert_eq(InventoryManager.gold_component.get_gold_amount(), -10)
```

Comment:
doesn't lose gold ensure you always have more than 0 ?

Reply:
No, I kinda let negative numbers be a thing in all of the components, but that's probably a bit dumb, so I'll change that

Reply:
good idea

Reply:
Ye, both torch and gold has helper functions like that, and now neither of them can naturaly get bellow 0

Reply:
okay, I think it would be good to have a `push_error` or `push_warning` at this place, because if we end up with gold below 0 there is an issue and we would need to fix it, so a little error or warning would be good for us

Reply:
I'll make them warnings, as I don't think it's bad enough to throw an error, but it's worth noteing